File: NextDNSProtector/app/src/main/java/com/nextdns/protector/BootReceiver.java
package com.nextdns.protector;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.widget.Toast;

/**
 * Boot Receiver - Reapplies DNS settings on device boot
 */
public class BootReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction()) ||
            "android.intent.action.QUICKBOOT_POWERON".equals(intent.getAction())) {

            // Reapply DNS settings after boot
            Intent serviceIntent = new Intent(context, DnsConfigService.class);
            serviceIntent.setAction("APPLY_DNS");
            context.startService(serviceIntent);

            Toast.makeText(context, "NextDNS settings reapplied after boot",
                          Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/nextdns/protector/receiver/BootReceiver.kt
package com.nextdns.protector.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import com.nextdns.protector.data.PreferencesManager
import com.nextdns.protector.service.NextDNSVpnService
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (intent.action == Intent.ACTION_BOOT_COMPLETED ||
            intent.action == "android.intent.action.QUICKBOOT_POWERON"
        ) {
            CoroutineScope(Dispatchers.IO).launch {
                val prefsManager = PreferencesManager(context)
                if (prefsManager.isDnsEnabled()) {
                    // Restart VPN service if it was enabled
                    val vpnIntent = Intent(context, NextDNSVpnService::class.java)
                    context.startService(vpnIntent)
                }
            }
        }
    }
}
